agent/structs/config_entry_inline_certificate.go

```diff
@@ -64,6 +64,30 @@ func (e *InlineCertificateConfigEntry) Validate() error {
 	return nil
 }
 
+func (e *InlineCertificateConfigEntry) Hosts() ([]string, error) {
+	certificateBlock, _ := pem.Decode([]byte(e.Certificate))
+	if certificateBlock == nil {
+		return nil, errors.New("failed to parse certificate PEM")
```

Comment:
would it make sense to use a sentinel error here so callers can use `errors.Is` to take action depending on the error type?

Reply:
I can do that in a follow-up. These are generic enough errors where we're pretty much only propagating this error back to the user or logging it and moving on, so I don't envision the need for ever trying to reflect on this or some wrapped version of it, but seeing as it's a static error string, I should probably make it into a variable that can be reused.